NuGet/NuGet.Client
# Clean-up a few unnecessary test projects

## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/206
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details:

 While doing NuGet/NuGet.Client#3289, I noticed that there are a bunch of unnecessary projects/extra projects.
This is a list of things that can be cleaned up.

* remove NuGet.Common.FuncTest 
  * This is a func test project with no tests
* clean-up NuGet.Core.FuncTest 
  * These are probably just leftover tests from the NuGet.Core split into 10ish different projects. All the tests are better suited in a different test project so I've moved them there.
* remove utility for generating licenses from solution.
  * There is no benefit to this being in the solution.
* remove utility for generating end to end test packages from solution.
  * Same as above, these are tests with no dependencies and are only built sporadically. We can reconsider whether these even belong in the test extensions folder or if they should be in a utility one. 
## Testing/Validation

Tests Added: No  
Reason for not adding tests:  Mostly moving around things :) 
Validation:
